### ISSUE TITLE
Bad performance in the admin panel while picking a relation

### ISSUE THREAD

**Issue by C4Grey** [bolt/bolt#4862]
Another "big data"-related issue. I have a few tables with 30-100K entries(this is really not so much comparing to other projects), and there is a one-to-many relation between those tables. Guess what happens when I am trying to pick a relation from a drop-down list? Correct, it loads a hell of a record, everything freezes, browser crashes, chaos, anger, suffering...It seems like there should be some full-screen pop-up(or even another page because modal windows are evil) providing a rich possibilities of selection of a required relation instead of a uninformative drop-down which is applicable only for short lists.

**Comment by GawainLynch** [comment]
Wow… you certainly get the prize for pushing Bolt hard! :grin:

**Comment by bobdenotter** [comment]
@C4Grey This is one of those typical things that's been in the back of my mind for ages, but never had the time to "fix" it. If you'd be willing to help engineer a fix, I'm sure a lot of people would benefit from that as well. :-)

**Comment by SahAssar** [comment]
I've been experimenting with just having select2 have the data in JSON instead of in the actual select field, which works very well when the editcontent page is loaded, but of course it still takes a long time to do the initial load. I've tried this on a relation to a contenttype with 166244 records. It might be an okay intermediate solution for sites that have less than 10k or so entries to relate to, but we really want to AJAXify this, right?

This is what I did for JSONifying the select2 field: https://gist.github.com/SahAssar/74ef4d67363b4d53edc7
